pojo/src/main/java/com/connexta/transformation/pojo/RequestInfoPojo.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.pojo;
+
+import com.connexta.transformation.pojo.unknown.UnknownRequestInfoPojo;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.net.URL;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * This class provides a pojo implementation for the request information associated with a
+ * particular transformation capable of reloading all supported fields for all supported versions
+ * from persistence. It also provides the capability of persisting back the fields based on the
+ * latest version format.
+ */
+@JsonPropertyOrder({"id", "version", "finalLocation", "currentLocation", "metadataLocation"})
```

Comment:
❓ Do we need to include `clazz` as a property here?

Reply:
The `JsonPropertyOrder` is not  a required annotation, it just makes it more easily readable when testing. You are also not required to define all properties. The one listed will be put first in the generated Json and the rest added after that in whatever order Jackson sees fit. I didn't bother about `clazz`, although I could certainly add it. On the other side, I never tested if Jackson will like it if I list `clazz` which is not a property of the object but one it, itself, adds automatically at some point during its serialization process. 

Let me know if you feel strongly about it and I can test it out.

Reply:
Now that I am going through others, I am noticing I did add it in other places so I'll do the same here.